lib/Target/SubstraitPB/Export.cpp

```diff
@@ -833,6 +856,29 @@ SubstraitExporter::exportOperation(LiteralOp op) {
   // `TimeType`.
   else if (auto timeType = dyn_cast<TimeType>(literalType)) {
     literal->set_time(mlir::cast<TimeAttr>(value).getValue());
+  }
+  // `IntervalType`'s.
+  else if (auto timeType = dyn_cast<IntervalYearMonthType>(literalType)) {
+    auto intervalYearToMonth = std::make_unique<
+        ::substrait::proto::Expression_Literal_IntervalYearToMonth>();
+    auto intervalYear =
+        mlir::cast<IntervalYearMonthAttr>(value).getYearsValue();
+    auto intervalMonth =
+        mlir::cast<IntervalYearMonthAttr>(value).getMonthsValue();
```

Comment:
Nit: use `auto` only if the type appears in the RHS of a cast or similar, see [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable). Also, my change factors out the cast, which makes it more readable and shorter. (Be sure to check if `int32_t` is the type returned by `getXValue()`.)
```suggestion
    auto intervalYearMonth = mlir::cast<IntervalYearMonthAttr>(value);
    int32_t intervalYear = intervalYearMonth.getYearsValue();
    int32_t intervalMonth = intervalYearMonth.getMonthsValue();
```